### ISSUE TITLE
Search corrects spelling without telling the user

### ISSUE THREAD

**Issue by astrojuanlu** [readthedocs/readthedocs.org#8534]
I searched for "context" but started getting results for "content", which was confusing.

![Screenshot 2021-09-28 at 13-47-58 Search — Read the Docs 5 24 0 documentation](https://user-images.githubusercontent.com/316517/135081107-b5c845b7-a722-4e5a-b0fa-bb7b2abcb045.png)

**Comment by stsewd** [comment]
This is because our project has the fuzzy search flag enabled. We aren't checking for spelling errors, we are just doing a fuzzy search if you search a "single term". You can use `"context"` to get an exact search.

**Comment by stsewd** [comment]
This was actually so people aren't frustrated if they made a typo or don't know the exact word. This is controlled via https://github.com/readthedocs/readthedocs.org/blob/f0da2a478705c794ce046d67a3eededf5595fd33/readthedocs/search/faceted_search.py#L178-L184 https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness

We were also exploring into having a UI for displaying the search options. So, not sure, personalty I like the default fuzzy and prefix search, and use `"` if I need to search something specific (rarely I need to use quotes really)

**Comment by astrojuanlu** [comment]
I see how users might get frustrated if they made a typo, or if they used a different spelling (British vs American for example). So doing the fuzzy search is a great addition.

However, for cases in which the user _didn't_ make a typo:

1) It is weird to see matches of words you're not interested in, and
2) **There is no way to discover how to look for an exact match** (yes, using double quotes)

So, I think it would be cool to have an admonition on top of the search page saying something like

> Performing a fuzzy match on the backend, use `"{term}"` to do an exact match. Read more about [the search syntax](https://docs.readthedocs.io/en/stable/guides/searching-with-readthedocs.html) and [how Read the Docs enhances search on your project](https://docs.readthedocs.io/en/stable/server-side-search.html).

**Comment by stsewd** [comment]
+1 on adding that admonition in our search extension

**Comment by stsewd** [comment]
We could also inject that message in search page without requiring our search extension, but not sure about changing that content in behalf of users.

**Comment by humitos** [comment]
I think **`search.html` vs modal** solves different UX issues --and both are good! 👍🏼 

Adding it on the modal communicates the extra options _before_ the user searches for a term. Also, if the user doesn't hit enter on that modal, they will never see the `search.html`  page.

On the `search.html` works for the case that the user searches for `content` and then realize that it's showing `context` results as well, so they want to force it to be exact _after_ searching for the term.

In any case, I think it would also be good to have an icon or something in the search box in the left (in our theme) that can hover/click and go directly to the documentation.

**Comment by nienn** [comment]
> Performing a fuzzy match on the backend, use "{term}" to do an exact match. Read more about the search syntax and how Read the Docs enhances search on your project.

I would also like to see the words that are being searched appear somewhere in a more intuitive way.
Maybe each word could even have an (x) nearby that the user could click to exclude that word?

**Comment by astrojuanlu** [comment]
I don't know how the fuzzy search works, but maybe we don't have the list of "extra words" that are being searched? I'll let others clarify that though.

**Comment by stsewd** [comment]
@astrojuanlu yeah, not sure if we have the list (but we can extract them from the matches), but we also do a prefix search (words that start with the current search), so the list could get pretty large.

**Comment by nienn** [comment]
> but we also do a prefix search (words that start with the current search), so the list could get pretty large.

We don't need to display the full list, just the 1st N search terms.

**Comment by humitos** [comment]
This would be a really good UI element (`Fuzzy` checkbox) in the search as you type addon. I'm moving this issue over there.

> Performing a fuzzy match on the backend, use `"{term}"` to do an exact match. Read more about [the search syntax](https://docs.readthedocs.io/en/stable/guides/searching-with-readthedocs.html) and [how Read the Docs enhances search on your project](https://docs.readthedocs.io/en/stable/server-side-search.html).

This is tracked in https://github.com/readthedocs/addons/issues/33
